include/pro/pro.hpp

```diff
@@ -114,46 +114,30 @@ class ProWrapper : public Napi::ObjectWrap<ProWrapper> {
+            assertIsNumber(first.Get("requestVersion"), "proProofRequestBody.requestVersion");
+            assertIsNumber(first.Get("unixTsMs"), "proProofRequestBody.unixTsMs");
+            auto requestVersion = first.Get("requestVersion").As<Napi::Number>();
+            auto unix_ts_ms = toCppSysMs(first.Get("unixTsMs"), "proProofRequestBody.unixTsMs");
 
-            assertIsUInt8Array(first.Get("master_privkey"), "proProofRequestBody.master_privkey");
-            assertIsUInt8Array(
-                    first.Get("rotating_privkey"), "proProofRequestBody.rotating_privkey");
+            assertIsUInt8Array(first.Get("masterPrivkey"), "proProofRequestBody.masterPrivkey");
+            assertIsUInt8Array(first.Get("rotatingPrivkey"), "proProofRequestBody.rotatingPrivkey");
 
-            // stack allocate to the buffer view so the ref doesnt get deleted
-            auto master_privkey_napi = first.Get("master_privkey");
-            auto rotating_privkey_napi = first.Get("rotating_privkey");
+            auto master_privkey_js = first.Get("masterPrivkey");
+            auto rotating_privkey_js = first.Get("rotatingPrivkey");
             auto master_privkey =
-                    toCppBufferView(master_privkey_napi, "proProofRequestBody.master_privkey");
+                    toCppBuffer(master_privkey_js, "proProofRequestBody.masterPrivkey");
             auto rotating_privkey =
-                    toCppBufferView(rotating_privkey_napi, "proProofRequestBody.rotating_privkey");
+                    toCppBuffer(rotating_privkey_js, "proProofRequestBody.rotatingPrivkey");
 
-            assert(master_privkey.size() == 64);
-            assert(rotating_privkey.size() == 64);
+            assert_length(master_privkey, 64, "master_privkey");
+            assert_length(rotating_privkey, 64, "rotating_prevkey");
 
-            pro_backend::GetProProofRequest proProofRequest{
-                    .version = static_cast<uint8_t>(version.Int32Value()),
-                    .unix_ts = unix_ts,
-            };
-
-            auto [master_sig, rotating_sig] = proProofRequest.build_sigs(
-                    proProofRequest.version,
+            auto json = pro_backend::GetProProofRequest::build_to_json(
+                    static_cast<uint8_t>(requestVersion.Int32Value()),
                     master_privkey,
                     rotating_privkey,
-                    proProofRequest.unix_ts);
-
-            assert(master_sig.size() == 64);
-            assert(rotating_sig.size() == 64);
-
-            proProofRequest.master_sig = master_sig;
-            proProofRequest.rotating_sig = rotating_sig;
-            memcpy(proProofRequest.master_pkey.data(), master_privkey.data(), 32);
-            memcpy(proProofRequest.rotating_pkey.data(), rotating_privkey.data(), 32);
+                    unix_ts_ms);
 
-            auto json = proProofRequest.to_json();
             auto json_str = Napi::String::New(env, json);
             return json_str;
```

Comment:
this isn't needed, you can just `return json` here, `wrapResult` on line 99 will wrap it for you